src/core/simulation.jl

```diff
@@ -425,10 +425,15 @@ function initial_condition_update!(
     ini_cond_vector = get_initial_conditions(stage.internal.psi_container)[ini_cond_key]
     for ic in get_initial_conditions(stage.internal.psi_container)[ini_cond_key]
         name = device_name(ic)
-        interval_chronology = get_stage_interval_chronology(sim, stage.name)
+        interval_chronology =
+            get_stage_interval_chronology(sim.sequence, get_stage_name(sim, stage))
         var_value =
             get_stage_variable(interval_chronology, (stage => stage), name, ic.update_ref)
-        cache = get_cache(stage, ic.cache_key)
+        if isnothing(ic.cache_type)
```

Comment:
why this here? the get cache should return a nothing if you run the get_cache function.

Reply:
sorry, about that

Reply:
Well this function only takes `<: AbstractCache ` but `ic.cache_type` can be a nothing which breaks the that call and this is how you have it for `InterStageChronology`

Reply:
ok, just leave it like this. I need to do some refactoring of these updates to make the code cleaner.